modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/SkuResourceImpl.java

```diff
@@ -406,28 +406,14 @@ private Sku _updateSKU(CPInstance cpInstance, Sku sku) throws Exception {
 						contextUser.getTimeZone()),
 					cpInstance.getDisplayDate())));
 
-		int expirationDateMonth = 0;
-		int expirationDateDay = 0;
-		int expirationDateYear = 0;
-		int expirationDateHour = 0;
-		int expirationDateMinute = 0;
-
-		if (cpInstance.getExpirationDate() != null) {
-			DateConfig expirationDateConfig = new DateConfig(
-				DateConfigUtil.convertDateToCalendar(
-					GetterUtil.getDate(
-						sku.getExpirationDate(),
-						DateFormatFactoryUtil.getDate(
-							contextAcceptLanguage.getPreferredLocale(),
-							contextUser.getTimeZone()),
-						cpInstance.getExpirationDate())));
-
-			expirationDateMonth = expirationDateConfig.getMonth();
-			expirationDateDay = expirationDateConfig.getDay();
-			expirationDateYear = expirationDateConfig.getYear();
-			expirationDateHour = expirationDateConfig.getHour();
-			expirationDateMinute = expirationDateConfig.getMinute();
-		}
+		DateConfig expirationDateConfig = DateConfig.toExpirationDateConfig(
```

Comment:
What if both `sku.getExpirationDate()` and `cpInstance.getExpirationDate()` are `null`?

Reply:
@alessiorendina  That's why in the original PR, i changed the if check on line 415 to (sku.getExpirationDate() != null).
@andrea-ale-sbarra Should we put the if condition back?

Reply:
if both are null it is set the now date +1 month. basically it's applying the same pattern of the add.